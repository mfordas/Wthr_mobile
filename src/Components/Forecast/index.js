import React from 'react';
import { View, Text, Image, ScrollView } from 'react-native';
import { connect } from 'react-redux';

import { chooseIcon } from '../../Common_Functions/icon_choose';
import { changeMetersPerSecondToKMetersPerHour } from '../../Common_Functions/change_m_to_km';
import mainStyling from '../../main_styling/main_styling';

import gaugeSrc from '../../img/gauge.png';
import humiditySrc from '../../img/humidity.png';
import windSrc from '../../img/wind.png';

class Forecast extends React.Component {
  constructor(props){
    super(props)
  }

  dateFormat = (date) => {
    const dateToFormat = new Date(date*1000);
    const day = dateToFormat.getDate();
    const month = dateToFormat.getMonth()+1;
    return <Text style={mainStyling.conditionsText}>{day}/{month}</Text>
  }

  render() {
    return (
              <ScrollView style={{flex:1, flexDirection:'row'}} contentContainerStyle={{flexGrow: 1, justifyContent: 'space-between'}} horizontal={true}>
                  {this.props.weatherData.weatherData.daily.map(
                      day => { 
                    return <View style={mainStyling.forecastContainer}>
                                <View style={mainStyling.conditionsContainerVertical}>
                                  <Text style={mainStyling.conditionsText}>Date</Text>
                                  {this.dateFormat(day.sunrise)}
                                </View>
                                <View style={mainStyling.conditionsContainerVertical}>
                                <Image style={mainStyling.conditionsIcon} source={chooseIcon(day.weather[0].icon)} />
                                </View>
                                <View style={mainStyling.conditionsContainerVertical}>
                                <Text style={mainStyling.conditionsText}>{Math.round(day.temp.day)}Â°C</Text>
                                </View>
                                <View style={mainStyling.conditionsContainerVertical}>
                                <Image style={mainStyling.conditionsIcon} source={windSrc} />
                                <Text style={mainStyling.conditionsText}>{changeMetersPerSecondToKMetersPerHour(day.wind_speed)} km/h</Text>
                                </View>
                                <View style={mainStyling.conditionsContainerVertical}>
                                <Image style={mainStyling.conditionsIcon} source={humiditySrc} />
                                <Text style={mainStyling.conditionsText}>{day.humidity}%</Text>
                                </View>
                                <View style={mainStyling.conditionsContainerVertical}>
                                <Image style={mainStyling.conditionsIcon} source={gaugeSrc} />
                                <Text style={mainStyling.conditionsText}>{day.pressure}hPa</Text>
                                </View>
                            </View>
                      }
                  )
                }
        </ScrollView> 
    );
  }
}

const mapStateToProps = state => ({
    weatherData: state.weatherData,
})

export default connect(mapStateToProps, { })(Forecast);