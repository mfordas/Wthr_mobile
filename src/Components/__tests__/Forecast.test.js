import 'react-native';
import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';

import Forecast from '../Forecast';
import mainStyling from '../../main_styling/main_styling';
import store from '../../store/store';

test('renders correctly', () => {
    const tree = renderer.create(<Provider store={store}><Forecast /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });