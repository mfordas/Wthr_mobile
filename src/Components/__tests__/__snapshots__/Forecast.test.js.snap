// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
      "justifyContent": "space-between",
    }
  }
  horizontal={true}
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View />
</RCTScrollView>
`;
